### PR TITLE
filter1D cursor forwarding example, and some fixes in pretty printing

### PR DESCRIPTION
**Pull request by yamaguchi1024**
Pretty printing is exposed to Cursor API

### CODE REVIEW

File: src/exo/LoopIR_pprint.py [hunk 1]
```diff
@@ -677,7 +677,8 @@ def _print_cursor_stmt(
 
     else:
         lines = _print_stmt(stmt, env, indent)
-    if cur == target:
+
+    if isinstance(target, Node) and isinstance(cur, Node) and cur._node == target._node:
```
Comment (alexreinking) [inline src/exo/LoopIR_pprint.py:681]: Why was this changed? Is there a test case that demonstrates a need for this?

**Review by alexreinking — CHANGES_REQUESTED** [review]

File: src/exo/LoopIR_pprint.py (hunk 1 above)
Comment (yamaguchi1024) [reply src/exo/LoopIR_pprint.py:681]: cur and target were always not equal even though they were pointing to the same node. So "# <-- NODE" was never added to the output code.

Comment (yamaguchi1024) [reply src/exo/LoopIR_pprint.py]: @alexreinking Sorry, my bad. I thought this line was failing the test I wanted to add, but it wasn't related. I think this was just some leftover from debugging. Thanks for catching this!
